tests/coreos-install/util/util.go

```diff
@@ -106,14 +115,43 @@ func FetchLocalImage(t *testing.T) string {
 	return tmpDir
 }
 
-func DownloadFile(tmpDir, name string) error {
+// Used to get defaults for channel, board, & version, first checks if the
+// host machine is Container Linux and if so uses the data from the machine
+// otherwise defaults to stable, amd64-usr, & current respectively
+func GetDefaultChannelBoardVersion(t *testing.T) (string, string, string) {
+	data, err := ioutil.ReadFile("/usr/lib/os-release")
+	if err != nil {
+		return "stable", "amd64-usr", "current"
+	}
+
+	os, err := TryRegexpSearch("id", "ID=['\"]?([A-Za-z0-9 \\._\\-]*)['\"]?", data)
+	if err != nil || os != "coreos" {
+		return "stable", "amd64-usr", "current"
+	}
+
+	version := RegexpSearch(t, "version", "VERSION_ID=['\"]?([A-Za-z0-9 \\._\\-]*)['\"]?", data)
+	board := RegexpSearch(t, "board", "COREOS_BOARD=['\"]?([A-Za-z0-9 \\._\\-]*)['\"]?", data)
+
+	data, err = ioutil.ReadFile("/etc/coreos/update.conf")
+	if err != nil {
+		t.Fatalf("reading /etc/coreos/update.conf: %v", err)
+	}
+
+	channel := RegexpSearch(t, "channel", "GROUP=['\"]?([A-Za-z0-9 \\._\\-]*)['\"]?", data)
+
+	return channel, board, version
+}
+
+func DownloadFile(t *testing.T, tmpDir, name string) error {
 	file, err := os.Create(filepath.Join(tmpDir, name))
 	if err != nil {
 		return fmt.Errorf("failed to create file: %v", err)
 	}
 	defer file.Close()
 
-	resp, err := http.Get(fmt.Sprintf("https://stable.release.core-os.net/amd64-usr/current/%s", name))
+	channel, board, version := GetDefaultChannelBoardVersion(t)
```

Comment:
Just noticed this: a function that fails with `t.Fatal` shouldn't be called by a function that returns `error`, since a caller of the outer function will think all errors can be caught that way.